tt_metal/impl/buffers/semaphore.cpp

```diff
@@ -8,24 +8,32 @@ namespace tt {
 
 namespace tt_metal {
 
-Semaphore::Semaphore(const Semaphore &other) : core_range_set_(other.core_range_set_), address_(other.address_), initial_value_(other.initial_value_), core_type_(other.core_type_) {}
```

Comment:
beyond the scope of this work, but I wonder if we want to delete these copy/move constructors